main.cpp

```diff
@@ -1,84 +1,52 @@
-            MajorColor getMajor() {
-                return majorColor;
-            }
-            MinorColor getMinor() {
-                return minorColor;
-            }
-            std::string ToString() {
-                std::string colorPairStr = MajorColorNames[majorColor];
-                colorPairStr += " ";
-                colorPairStr += MinorColorNames[minorColor];
-                return colorPairStr;
-            }
-    };
-
-    ColorPair GetColorFromPairNumber(int pairNumber) {
-        int zeroBasedPairNumber = pairNumber - 1;
-        MajorColor majorColor = 
-            (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
-        MinorColor minorColor =
-            (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
-        return ColorPair(majorColor, minorColor);
-    }
-    int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
-        return major * numberOfMinorColors + minor + 1;
-    }
+	TelCoColorCoder::CColorPair colorPair =
+		TelCoColorCoder::GetColorFromPairNumber(pairNumber);
+	std::cout << "Got pair" << colorPair.ColorPairToString() << std::endl;
+	assert(colorPair.GetMajor() == expectedMajor);
+	assert(colorPair.GetMinor() == expectedMinor);
 }
 
-void testNumberToPair(int pairNumber,
-    TelCoColorCoder::MajorColor expectedMajor,
-    TelCoColorCoder::MinorColor expectedMinor)
+/**
+ *  Func Description: testcase to verify match in major and minor color
+ *  with obtained code.
+ */
+void TestPairToNumber(
+	TelCoColorCoder::MajorColor major,
+	TelCoColorCoder::MinorColor minor,
+	int expectedPairNumber)
 {
-    TelCoColorCoder::ColorPair colorPair =
-        TelCoColorCoder::GetColorFromPairNumber(pairNumber);
-    std::cout << "Got pair " << colorPair.ToString() << std::endl;
-    assert(colorPair.getMajor() == expectedMajor);
-    assert(colorPair.getMinor() == expectedMinor);
+	int pairNumber = TelCoColorCoder::GetPairNumberFromColor(major, minor);
+	std::cout << "Got pair number " << pairNumber << std::endl;
+	assert(pairNumber == expectedPairNumber);
 }
 
-void testPairToNumber(
-    TelCoColorCoder::MajorColor major,
-    TelCoColorCoder::MinorColor minor,
-    int expectedPairNumber)
+/**
+ *  Func Description: Test api to get the color code form
+ */
+void TestRefferenceManual() 
 {
-    int pairNumber = TelCoColorCoder::GetPairNumberFromColor(major, minor);
-    std::cout << "Got pair number " << pairNumber << std::endl;
-    assert(pairNumber == expectedPairNumber);
-}
+	TelCoColorCoder::PrintColorCodesForm();
+ }
 
 int main() {
-    testNumberToPair(4, TelCoColorCoder::WHITE, TelCoColorCoder::BROWN);
-    testNumberToPair(5, TelCoColorCoder::WHITE, TelCoColorCoder::SLATE);
+	TestNumberToPair(4, TelCoColorCoder::WHITE, TelCoColorCoder::BROWN);
+	TestNumberToPair(5, TelCoColorCoder::WHITE, TelCoColorCoder::SLATE);
 
-    testPairToNumber(TelCoColorCoder::BLACK, TelCoColorCoder::ORANGE, 12);
-    testPairToNumber(TelCoColorCoder::VIOLET, TelCoColorCoder::SLATE, 25);
```

Comment:
This file is all about testing. Should it still be called main.cpp?

Reply:
Thanks for the feedback. I have updated as per your comment.